### PR TITLE
chore: Update Jest to v29

### PR DESCRIPTION
**Pull request by legobeat**
_This is #7160 rebased on `main` with changes to snapshots reapplied_

---
Jest has been updated to v29, along with related dependencies. This change comes with a number of new features, such as the new `jest.replaceProperty()` function.

The only breaking change affecting us is a change to the snapshot format. All snapshots have been updated.

See here for the migration guide: https://jest-archive-august-2023.netlify.app/docs/upgrading-to-jest29/

See here for a full list of changes: https://github.com/jestjs/jest/blob/main/CHANGELOG.md#2900

#### Blocked by
- [x] #9459 

#### Blocking
- #9452

#### Previously 
- #7160
- #6277 
- #6876 
- #8960 

## **Pre-merge author checklist**

- [x] I’ve followed [MetaMask Coding Standards](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've completed the PR template to the best of my ability
- [x] I’ve included tests if applicable
- [x] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [x] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

**Comment by legobeat** [comment]
@SocketSecurity ignore npm/create-jest@29.7.0

new author ok

File: package.json [hunk 1]
```diff
@@ -377,7 +377,7 @@
     "@types/eth-url-parser": "^1.0.0",
     "@types/i18n-js": "^3.8.4",
     "@types/is-url": "^1.2.30",
-    "@types/jest": "^28.1.8",
+    "@types/jest": "^29.5.4",
```
Comment (leotm) [inline package.json:380]:
```suggestion
    "@types/jest": "^29.5.12",
```

but prob don't want to apply this suggestion, as it will miss updating yarn.lock

or can just bump this later

**Review by leotm — DISMISSED** [review]

File: package.json (hunk 1 above)
Comment (legobeat) [reply package.json]: nbd!

**Review by leotm — DISMISSED** [review]

File: app/lib/ppom/ppom.html.js
Comment (Gudahtt) [inline app/lib/ppom/ppom.html.js:1]: Any idea what the changes are here? This doesn't appear to be related to snapshots

Comment (legobeat) [reply app/lib/ppom/ppom.html.js:1]: Broken out here (effect of associated babel upgrade): #9458

I find it really quite unfortunate that this minified mess is checked in the way it is - it's practically unauditable...

Comment (leotm) [reply app/lib/ppom/ppom.html.js:1]: rendering the diff significantly slowed down my machine 🙈

image

Comment (legobeat) [reply app/lib/ppom/ppom.html.js:1]: - https://github.com/MetaMask/metamask-mobile/pull/9475

**Review by Gudahtt — DISMISSED** [review]
LGTM!

**Review by Gudahtt — APPROVED** [review]

**Review by Cal-L — APPROVED** [review]
Code LGTM, let's wait until e2e passes to merge
